frappe/public/js/frappe/form/controls/geolocation.js

```diff
@@ -44,52 +46,49 @@ frappe.ui.form.ControlGeolocation = class ControlGeolocation extends frappe.ui.f
 			this.map.zoomControl.remove();
 		} else {
 			this.bind_leaflet_draw_control();
+			this.bind_leaflet_event_listeners();
 			this.bind_leaflet_locate_control();
-			this.bind_leaflet_refresh_button();
+			this.bind_leaflet_data(value);
 		}
-		this.map.setView(frappe.utils.map_defaults.center, frappe.utils.map_defaults.zoom);
 	}
 
-	format_for_input(value) {
-		if (!this.map) return;
-		// render raw value from db into map
+	bind_leaflet_data(value) {
+		/* render raw value from db into map */
+		if (!this.map || !value) {
+			return;
+		}
 		this.clear_editable_layers();
-		if (value) {
-			var data_layers = new L.FeatureGroup().addLayer(
-				L.geoJson(JSON.parse(value), {
-					pointToLayer: function (geoJsonPoint, latlng) {
-						if (geoJsonPoint.properties.point_type == "circle") {
-							return L.circle(latlng, { radius: geoJsonPoint.properties.radius });
-						} else if (geoJsonPoint.properties.point_type == "circlemarker") {
-							return L.circleMarker(latlng, {
-								radius: geoJsonPoint.properties.radius,
-							});
-						} else {
-							return L.marker(latlng);
-						}
-					},
-				})
-			);
-			this.add_non_group_layers(data_layers, this.editableLayers);
-			try {
-				this.map.fitBounds(this.editableLayers.getBounds(), {
-					padding: [50, 50],
-				});
-			} catch (err) {
-				// suppress error if layer has a point.
-			}
-			this.editableLayers.addTo(this.map);
-		} else {
-			this.map.setView(frappe.utils.map_defaults.center, frappe.utils.map_defaults.zoom);
+
+		const data_layers = new L.FeatureGroup().addLayer(
+			L.geoJson(JSON.parse(value), {
+				pointToLayer: function (geoJsonPoint, latlng) {
+					if (geoJsonPoint.properties.point_type == "circle") {
+						return L.circle(latlng, { radius: geoJsonPoint.properties.radius });
+					} else if (geoJsonPoint.properties.point_type == "circlemarker") {
+						return L.circleMarker(latlng, { radius: geoJsonPoint.properties.radius });
+					} else {
+						return L.marker(latlng);
+					}
+				},
+			})
+		);
+		this.add_non_group_layers(data_layers, this.editableLayers);
+		try {
+			this.map.fitBounds(this.editableLayers.getBounds(), {
+				padding: [50, 50],
+			});
+		} catch (err) {
+			// suppress error if layer has a point.
 		}
+		this.editableLayers.addTo(this.map);
 		this.map.invalidateSize();
```

Comment:
**Fix the for `invalidateSize` issue explained in https://github.com/frappe/frappe/pull/16561#issuecomment-1570225211** (sorry for the long suggestion, these fixes are dependent on the changes in this PR)

```suggestion
		this.editableLayers.addTo(this.map);
		this.fit_and_recenter_map();
```

Add the following function to the control:
```js
fit_and_recenter_map() {
		// Spread map across the wrapper, recenter and zoom w.r.t bounds
		try {
			this.map.invalidateSize();
			this.map.fitBounds(this.editableLayers.getBounds(), {
				padding: [50, 50],
			});
		} catch (err) {
			// suppress error if layer has a point.
		}
	}
```

Now in `section.js` > `collapse()` add this action for geolocation control:
```js
// refresh signature fields
this.fields_list.forEach((f) => {
	if (f.df.fieldtype == "Signature") {
		f.refresh();
	}
	
	// fix
	if (f.df.fieldtype == "Geolocation") {
		f.fit_and_recenter_map();
	}
});
```